app/src/main/java/com/muzima/view/forms/HTMLFormDataStore.java

```diff
@@ -1248,6 +1248,31 @@ public String getDerivedObservationByPatientUuidAndDerivedConceptId(String patie
         return createDerivedObsJsonArray(derivedObservations);
     }
 
+    @JavascriptInterface
+    public String getInterventionsDerivedObservationByPatientUuid(String patientUuid) throws JSONException, DerivedConceptController.DerivedConceptFetchException {
+        List<DerivedObservation> derivedObservations = new ArrayList<>();
+        List<DerivedObservation> derivedObservationList = new ArrayList<DerivedObservation>(0);
+        try {
+            derivedObservations = derivedObservationController.getDerivedObservationByPatientUuid(patientUuid);
```

Comment:
Instead of getting all and then looping, get the derived obs by patient uuid and derived concept uuid then add to the list and sort